Dart/src/com/jetbrains/lang/dart/analyzer/DartAnalysisServerService.java

```diff
@@ -657,6 +661,35 @@ public void run() {
     });
   }
 
+  @NotNull
+  public List<HoverInformation> analysis_getHover(@NotNull final String _filePath, final int offset) {
+    final String filePath = FileUtil.toSystemDependentName(_filePath);
+    final AtomicReference<List<HoverInformation>> result = new AtomicReference<List<HoverInformation>>();
+
+    final AnalysisServer server = myServer;
+    if (server == null) {
+      return HoverInformation.EMPTY_LIST;
+    }
+
+    final CountDownLatch latch = new CountDownLatch(1);
+    server.analysis_getHover(filePath, offset, new GetHoverConsumer() {
+      @Override
+      public void computedHovers(HoverInformation[] hovers) {
```

Comment:
Correct.
I don't know for sure if DAS actually ever returns more than one, but it could.